tests/unit/type/set.tcl

```diff
@@ -23,12 +23,8 @@ start_server {
         assert_encoding $type myset
 
         # coverage for objectComputeSize
-        if {$type == "listpack"} {
-            assert_morethan [r memory usage myset] 58
-        } elseif {$type == "hashtable"} {
-            assert_morethan [r memory usage myset] 6080
-        }
-
+        assert_range [r memory usage myset] 50 12000 ;# sum of the size of each element in $initelems
+                                                      # for listpack 59 and hashtable 6081
```

Comment:
maybe you misunderstood me. but i'm arguing that the sum of the size of the elements is the same on both.
based on the lowest theoretical consumption of the actual data without overhead, not based on any measurement you took.

i.e. there are 130 strings of 1 to 3 characters, (30*3 + 90*2 +10*1), so that's 280.
so i meant the thresholds can be 280 (can't be lower than that), and some factor to allow overhead (like *2 or *10).

now i understand that in listpack these are actually encoded as integers and they take even less characters.
so for all i care, the thresholds can be 130 (the memory consumption can't possibly take less than one byte per element), and 280+50*130 (i.e. allowing 50 bytes of overhead per item).
```suggestion
        assert_range [r memory usage myset] 130 7000 ;# can't be lower than one byte per item, or higher than 50 bytes overhead beyond the sum of the actual strings.
```

Reply:
listpack takes less than one byte per element? IIRC the minimum should be 2 bytes.
what am i missing?

Reply:
@hwware there is just one element in initelems(listpack).
see above
```
    for {set i 0} {$i < 130} {incr i} {
        lappend initelems(hashtable) [format "i%03d" $i]  <- just insert 130 elements into hashtable
    }
```

Reply:
Thank you @sundb, you saved my day! I was thinking about 130 elements but not noticed it created only for hashtable. Now it matches the size and the elements.

it's true in listpack takes 2 bytes. here is the example.

```
127.0.0.1:6379> sadd myset foo
(integer) 1
127.0.0.1:6379>
127.0.0.1:6379>
127.0.0.1:6379> memory usage myset
(integer) 59
127.0.0.1:6379> sadd myset foo 111
(integer) 1
127.0.0.1:6379> memory usage myset
(integer) 61
127.0.0.1:6379>

```
@oranagra I have updated the comment as you suggested and updated the diff, could you please look at it.

Reply:
i think that now it's rather confusing (as evident by the history in this tread), that we have a comment for an assertion that's not encoding specific, and refers to the number of elements, where it's a different number of elements in each encoding.

i think we should either change the test to create the same amount of elements in either encoding (and maybe tune the config in order to control which encoding it used), or move the MEMORY USAGE test to another new test that we'll create which has the same number of elements in either encoding.

Reply:
Tried changing config set-max-listpack-entries to 130, but its getting conflicted with overrides in the starting of the script, so created new test for memory usage and added 128 elements for listpack and 129 for hashtable.